api/server.go

```diff
@@ -48,28 +50,58 @@ func NewServer(config common.Config) error {
 		return errors.New("'org' cannot be empty in the configuration")
 	}
 	Org = config.Org
-	repo := config.Repository
+	metadata := config.MetadataRepository
 
-	// Create metadata repository session
-	log.Debugf("Creating new MetadataRepository of type %s with configuration %+v (org: %s)", repo.Type, repo.Config, Org)
+	// Initialize metadata repository session
+	log.Debugf("Creating new session for MetadataRepository of type %s with configuration %+v (org: %s)", metadata.Type, metadata.Config, Org)
 
 	var metadataRepo dataset.MetadataRepository
 	var err error
-	switch repo.Type {
+	switch metadata.Type {
 	case "s3":
-		metadataRepo, err = s3metadatarepository.NewDefaultRepository(repo.Config)
+		if metadata.Config["prefix"] == nil {
+			metadata.Config["prefix"] = Org
+		} else {
+			metadata.Config["prefix"] = metadata.Config["prefix"].(string) + "/" + Org
+		}
+		metadataRepo, err = s3metadatarepository.NewDefaultRepository(metadata.Config)
 		if err != nil {
 			return err
 		}
 	default:
-		return errors.New("failed to determine metadata repository type, or type not supported: " + repo.Type)
+		return errors.New("failed to determine metadata repository type, or type not supported: " + metadata.Type)
 	}
 
-	// Create a shared session
-	for name, c := range config.Accounts {
-		log.Debugf("Creating new service for account '%s' with key '%s' in region '%s' (org: %s)", name, c.Akid, c.Region, Org)
+	// Create dataset service sessions
+	for name, a := range config.Accounts {
+		creds := map[string]interface{}{
+			"region":   a.Config.Region,
+			"akid":     a.Config.Akid,
+			"secret":   a.Config.Secret,
+			"endpoint": a.Config.Endpoint,
+		}
+		log.Debugf("Creating new service for account '%s' with key '%s' in region '%s' (org: %s, providers: %s)", name, creds["akid"], creds["region"], Org, a.StorageProviders)
+
+		dataRepos := make(map[string]dataset.DataRepository)
+
+		// initialize all supported data storage providers for each account
+		for _, p := range a.StorageProviders {
+			switch p {
+			case "s3":
+				dataRepo, err := s3datarepository.NewDefaultRepository(creds)
+				if err != nil {
+					return err
+				}
+				dataRepos["s3"] = dataRepo
```

Comment:
would different accounts ever have different s3 providers?  you might want to define your storage providers explicitly in the configuration and give them a "name" that you use here.

Reply:
it could go either way but i think its more likely that *most* accounts share a storage provider configuration (outside of the account) but some may need a special one defined.

Reply:
Not sure what that gets us besides standardizing on the names for those providers, there's no actual configuration there

Reply:
ok lets see how it goes 👍